eth-custodian/contracts/EthCustodianProxy.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8;
+import '@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol';
+import '@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import {EthCustodian} from './EthCustodian.sol';
+import {SelectivePausableUpgradable} from './SelectivePausableUpgradable.sol';
+
+contract EthCustodianProxy is
```

Comment:
@kiseln  Potentially, the implementation contract can be replaced with this contract.
Taking this into account, we need to reserve slots for the `ProofKeeper.sol`